istioctl/cmd/workload.go

```diff
@@ -0,0 +1,402 @@
+// Copyright Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	networkingv1alpha3 "istio.io/api/networking/v1alpha3"
+	clientv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	"istio.io/istio/istioctl/pkg/multicluster"
+	"istio.io/istio/pilot/pkg/serviceregistry/kube/controller"
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/config/mesh"
+	"istio.io/istio/pkg/config/schema/collections"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/kube/inject"
+	"istio.io/istio/pkg/util/gogoprotomarshal"
+)
+
+var (
+	tokenDuration  int64
+	name           string
+	serviceAccount string
+	filename       string
+	outputDir      string
+	clusterID      string
+	ports          []string
+)
+
+const (
+	filePerms = os.FileMode(0744)
+)
+
+func workloadCommands() *cobra.Command {
+	workloadCmd := &cobra.Command{
+		Use:   "workload",
+		Short: "Commands to assist in configuring and deploying workloads running on VMs and other non-k8s environments",
+	}
+	workloadCmd.AddCommand(groupCommand())
+	workloadCmd.AddCommand(entryCommand())
+	return workloadCmd
+}
+
+func groupCommand() *cobra.Command {
+	groupCmd := &cobra.Command{
+		Use:   "group",
+		Short: "Commands dealing with WorkloadGroup resources",
+	}
+	groupCmd.AddCommand(createCommand())
+	return groupCmd
+}
+
+func entryCommand() *cobra.Command {
+	entryCmd := &cobra.Command{
+		Use:   "entry",
+		Short: "Commands dealing with WorkloadEntry resources",
+	}
+	entryCmd.AddCommand(configureCommand())
+	return entryCmd
+}
+
+func createCommand() *cobra.Command {
```

Comment:
The more I think about it the less value I see in this command. It is not doing anything unique, and seems inconsistent we have it for WG but not for all the other Istio CRs (and I am not recommending we create it for those :slightly_smiling_face: ). But perfectly fine to keep it if others like it, its not adding a lot of cost